File: python_data_structures/01_linear_search.py
# Linear Search Algorithm


def linear_search(array, target):
    for value in array:
        if value == target:
            return True
    return False


if __name__ == "__main__":
    array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

    print("Array: " + str(array))

    target = int(input("Enter the target value: "))
    result = linear_search(array, target)

    print("Value found: " + str(result))

File: data_structures/cpp/src/01_linear_search.cpp
// Linear Search Algorithm

#include <iostream>

using namespace std;

void linear_search(int array[], int target, int max_size)
{
    for (int i = 0; i < max_size; i++)
    {
        if (array[i] == target)
        {
            cout << "Target value found!" << endl;
            break;
        }
    }
}

int main()
{
    int array[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    int max_size = sizeof(array) / sizeof(array[0]);
    int target = 5;

    // Implement linear search
    linear_search(array, target, max_size);

    return 0;
}